src/write/elf.rs

```diff
@@ -1075,6 +1115,7 @@ struct SectionHeader {
 }
 
 /// Native endian version of `Sym64`.
+#[repr(C)]
```

Comment:
This shouldn't be required.

Reply:
So, this part did feel particularly hacky, but I seemed to need to "cast" the data section as an array of `Sym` (otherwise the casting didn't create proper symbols).

Reply:
Oh, I didn't look at the code very closely... that unsafe code is wrong. You need to be using `elf::Sym32` or `elf::Sym64` instead. But as I said, it would be better if we did this differently (we build the `.dynsym` section ourselves).